scripts/list_updatable_packages

```diff
@@ -0,0 +1,67 @@
+#!/usr/bin/env python3
+
+import json
+import os
+import sys
+from collections import namedtuple
+from pathlib import Path
+from typing import Generator, Iterable, Mapping, Optional
+
+from debian.changelog import Changelog
+from debian.debian_support import NativeVersion
+
+import requests
+
+SESSION = requests.Session()
+
+
+Package = namedtuple('Package', ['package_name', 'version', 'gem_name', 'directory'])
+
+
+def find_packages() -> Generator[Package, None, None]:
+    for plugin_rb in Path('plugins/').glob('**/*.rb'):
+        gem_name = plugin_rb.read_text().strip().split()[1].strip("',\"")
+        package_changelog = next(plugin_rb.parent.glob('**/changelog'))
+        with package_changelog.open() as debchangelog:
+            changelog = Changelog(debchangelog)
+            version = NativeVersion(changelog.version).upstream_version
+            yield Package(str(changelog.package), version, gem_name, plugin_rb.parent.as_posix())
+
+
+def latest_version(spec: Package) -> str:
+    if spec.gem_name:
+        url = f'https://rubygems.org/api/v1/versions/{spec.gem_name}/latest.json'
+        response = SESSION.get(url)
+        response.raise_for_status()
+        return response.json()['version']
+    raise ValueError('Unable to determine latest version', spec)
+
+
+def build_matrix(specs: Iterable[Package]) -> Generator[dict, None, None]:
+    for spec in specs:
+        current_version = spec.version
+        new_version = latest_version(spec)
+        if current_version != new_version:
+            entry = {
+                'directory': spec.directory,
+                'package_name': spec.package_name,
+                'gem_name': spec.gem_name,
+                'current_version': current_version,
+                'new_version': new_version,
+            }
+            yield entry
+
+
+def main() -> None:
+    matrix = list(build_matrix(find_packages()))
+
+    if 'GITHUB_ACTION' in os.environ:
+        directories = [entry['directory'] for entry in matrix]
+        print(f'::set-output name=directories::{json.dumps(directories)}')
+        print(f'::set-output name=matrix::{json.dumps(matrix)}')
+    for entry in matrix:
+        print(entry['directory'], entry['new_version'])
```

Comment:
This should match what's called. In this case it's `./scripts/update_package.rb --name $package_name --version $version`. So `directory` here doesn't make a lot of sense?

Reply:
I didn't look too closely at that part (which is copied from you 1:1), but yeah, probably

Reply:
looking closer, the directory names in `plugins/` are *not* **always** package names (yay):
* `ruby-foreman-azure-rm` is a package name
* `smart_proxy_chef` is a gem name

So, really, one would need to call `./scripts/update_package.rb --name $(basename $directory) --version $version`

Reply:
Looks like that works